### ISSUE TITLE
abs documentation doesn't explain mapping behavior

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1269]
Imported from SourceForge on 2024-07-04 07:53:19
Created by **[creidieki](https://sourceforge.net/u/creidieki/)** on 2010-06-02 14:58:07
Original: https://sourceforge.net/p/maxima/bugs/2006

---

The documentation for the abs\(expr\) function doesn't mention that it maps over structures.  \(In my testing, it seems to map over lists but not matrices?\).

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 07:53:21
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2010-06-03 05:15:25
Original: https://sourceforge.net/p/maxima/bugs/2006/#bf12

---

abs automatically maps over lists, matrices, and equations.

\(%i81\) abs \(\[1, -2\]\);
\(%o81\)                              \[1, 2\]
\(%i82\) abs \(matrix \(\[1, -2\]\)\);
\(%o82\)                             \[ 1  2 \]
\(%i83\) abs \(1 = -2\);
\(%o83\)                               1 = 2

There isn't any way to discover that short of reading the source code \(which is how I figured it out\). That's a bug in the documentation, as pointed out by the original poster.

Aside from the documentation, there ought to be a way to easily discover the algebraic properties of functions. This mapping-over-objects is hard-wired in the code, it's not a declared property. It's not a bug since it works OK but still it would be better to have discoverable properties.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 07:53:24
Created by **[crategus](https://sourceforge.net/u/crategus/)** on 2010-06-04 17:14:54
Original: https://sourceforge.net/p/maxima/bugs/2006/#751c

---

I had a look at the implementation of the abs function.

I think we should cut out the mapping over bags from the simplifying function simpabs and put the property distribute\_over on the property list.

In general, we have the possibility to look up the properties of a function with the Maxima command properties. For the abs function we get

\(%i8\) properties\(abs\);
\(%o8\) \[rule, gradef\]

This is an example for the Bessel J function. We get much more information:

\(%i9\) properties\(bessel\_j\);
\(%o9\) \[conjugate function, complex characteristic, limit function, integral, 
gradef, distributes over bags, rule, noun, transfun\]

Some time ago we extended the function properties to show more correct the properties. We can even extend it and we should document the function properties better.

By the way:

I have detected that we support the symbol %mabs. This is a workaround to get the correct simplification of a noun form like 'abs\(-1\). But this is not necessary. The underlying problem is that the properties noun, verb, alias and reversealiase are not implemented correctly for the abs function. When we correct this we no longer need to support the symbol '%mabs.

Dieter Kaiser
